processor/logdedupprocessor/counter.go

```diff
@@ -79,14 +79,14 @@ func (l *logAggregator) Export(ctx context.Context) plog.Logs {
 }
 
 // Add adds the logRecord to the resource aggregator that is identified by the resource attributes
-func (l *logAggregator) Add(resource pcommon.Resource, scope pcommon.InstrumentationScope, logRecord plog.LogRecord) {
+func (l *logAggregator) Add(resource pcommon.Resource, scope pcommon.InstrumentationScope, logRecord plog.LogRecord, dedupFields []string) {
 	key := getResourceKey(resource)
 	resourceAggregator, ok := l.resources[key]
 	if !ok {
 		resourceAggregator = newResourceAggregator(resource)
 		l.resources[key] = resourceAggregator
 	}
-	resourceAggregator.Add(scope, logRecord)
+	resourceAggregator.Add(scope, logRecord, dedupFields)
```

Comment:
Instead of threading these fields through on each call to `Add`, could we move it to a property on the aggregators (resource, scope, log) stucts and set when creating the new instances?

Reply:
I updated the code, thanks for the feedback!